### ISSUE TITLE
MTS Division Issue?

### ISSUE THREAD

**Issue by kyleabeauchamp** [pandegroup/openmm#1730]
I'm wondering if we may have an issue with the MTS integrator:

```
        stepsPerParentStep = substeps//parentSubsteps
        if stepsPerParentStep < 1 or stepsPerParentStep != int(stepsPerParentStep):
            raise ValueError("The number for substeps for each group must be a multiple of the number for the previous group")
```

Doesn't `//` indicate floor division, which always returns an integer?  Thus, isn't the second test always true?

IMHO don't we want to do true division, run the integer check, then explicitly cast to int after the multiplicity check?

**Comment by kyleabeauchamp** [comment]
cc @mjw99

**Comment by peastman** [comment]
Good point.  Yes, the second check should really be `stepsPerParentStep != float(substeps)/parentSubsteps`.  Casting to a float is necessary on Python 2, but not on Python 3.

**Comment by mpharrigan** [comment]
You could alternatively use a future import https://docs.python.org/2/library/__future__.html
```python
from __future__ import division
```

**Comment by jchodera** [comment]
Would this change actually have any effect on properly-formulated force splittings? Or would this just help to catch improperly-formulated splittings?

**Comment by kyleabeauchamp** [comment]
Not 100% sure, but I *think* this just prevents improperly-formulated splittings, which previously would silently run with poor energy conservation, as was reported to me via email from @mjw99

**Comment by jchodera** [comment]
I've observed that even properly formulated MTS splittings don't seem to help with energy conservation, which I find incredibly puzzling. There may be a deeper issue here.

**Comment by kyleabeauchamp** [comment]
I too have observed cases where the MTS splitting does not provide the level of improvement that I would expect, but I have no smoking gun at the moment.

**Comment by peastman** [comment]
This is strictly about catching mistakes.  It won't affect the behavior of correct splittings.

**Comment by kyleabeauchamp** [comment]
Fixed in #1731
